src/libs/TransactionPreviewUtils.ts

```diff
@@ -0,0 +1,334 @@
+import truncate from 'lodash/truncate';
+import type {OnyxEntry, OnyxInputValue} from 'react-native-onyx';
+import type {LocaleContextProps} from '@components/LocaleContextProvider';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type * as OnyxTypes from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import {abandonReviewDuplicateTransactions, setReviewDuplicatesKey} from './actions/Transaction';
+import {convertToDisplayString} from './CurrencyUtils';
+import DateUtils from './DateUtils';
+import Navigation from './Navigation/Navigation';
+import type {PlatformStackRouteProp} from './Navigation/PlatformStackNavigation/types';
+import type {TransactionDuplicateNavigatorParamList} from './Navigation/types';
+import {getOriginalMessage, getReportAction, isMessageDeleted, isMoneyRequestAction} from './ReportActionsUtils';
+import {canEditMoneyRequest, isIOUReport, isPaidGroupPolicy, isPaidGroupPolicyExpenseReport, isReportApproved, isSettled} from './ReportUtils';
+import type {TransactionDetails} from './ReportUtils';
+import StringUtils from './StringUtils';
+import {
+    compareDuplicateTransactionFields,
+    getFormattedCreated,
+    hasMissingSmartscanFields,
+    hasNoticeTypeViolation,
+    hasPendingRTERViolation,
+    hasReceipt,
+    hasViolation,
+    hasWarningTypeViolation,
+    isAmountMissing,
+    isCardTransaction,
+    isCreatedMissing,
+    isDistanceRequest,
+    isFetchingWaypointsFromServer,
+    isMerchantMissing,
+    isOnHold,
+    isPending,
+    isPerDiemRequest,
+    isReceiptBeingScanned,
+} from './TransactionUtils';
+import ViolationsUtils from './Violations/ViolationsUtils';
+
+const emptyPersonalDetails: OnyxTypes.PersonalDetails = {
+    accountID: CONST.REPORT.OWNER_ACCOUNT_ID_FAKE,
+    avatar: '',
+    displayName: undefined,
+    login: undefined,
+};
+
+const chooseIdBasedOnAmount = (amount: number, negativeId: number, positiveId: number) => (amount < 0 ? negativeId : positiveId);
+
+function getIOUData(managerID: number, ownerAccountID: number, reportID: string | undefined, personalDetails: OnyxTypes.PersonalDetailsList | undefined, amount: number) {
+    const fromID = chooseIdBasedOnAmount(amount, managerID, ownerAccountID);
+    const toID = chooseIdBasedOnAmount(amount, ownerAccountID, managerID);
+
+    return {
+        from: personalDetails ? personalDetails[fromID] : emptyPersonalDetails,
+        to: personalDetails ? personalDetails[toID] : emptyPersonalDetails,
+        isIOU: reportID ? isIOUReport(reportID) : false,
+    };
+}
+
+const navigateToReviewFields = (
```

Comment:
instead of creating a function that is coupled to `Navigation` you could rewrite this function to return `route` to navigate to and backTo param.
Then Change the name to something like `getReviewNavigationRoute` (or a similar name, feel free to change it) and then just call navigate once in the place that this is used
What do you think?